### PR TITLE
Improve CREATE TYPE statement

### PR DESCRIPTION
**Pull request by pplam**
Improved the CREATE TYPE statement according to the grammar at https://www.postgresql.org/docs/current/sql-createtype.html

### CODE REVIEW

**Comment by m-novikov** [comment]
Please install `pre-commit` to auto-format grammar.js on commit.
See https://github.com/m-novikov/tree-sitter-sql#development
Also tests need updating

**Comment by pplam** [comment]
OK, thank for your suggestions m-novikov!

File: grammar.js [hunk 1]
```diff
@@ -1035,6 +1115,10 @@ function commaSep1(rule) {
   return sep1(rule, ",");
 }
 
+function commaSep(rule) {
```
Comment (m-novikov) [inline grammar.js:1118]: I am not sure I'll probably rewrite it as `optional(commaSep1(...))` in the future.

File: grammar.js [hunk 2]
```diff
@@ -557,8 +566,81 @@ module.exports = grammar({
...
+              ["INPUT", $._identifier],
+              ["OUTPUT", $._identifier],
+              ["RECEIVE", $._identifier],
+              ["SEND", $._identifier],
+              ["TYPMOD_IN", $._identifier],
+              ["TYPMOD_OUT", $._identifier],
+              ["ANALYZE", $._identifier],
+              ["SUBSCRIPT", $._identifier],
+              ["INTERNALLENGTH", choice($.number, kw("VARIABLE"))],
+              ["PASSEDBYVALUE", null],
```
Comment (m-novikov) [inline grammar.js:628]: Not a big fan of the special case with null, probably move it out of this map

**Pull request merged**